python/pyspark/sql/tests/connect/test_connect_function.py

```diff
@@ -0,0 +1,184 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Any
+import unittest
+import tempfile
+
+from pyspark.testing.sqlutils import have_pandas, SQLTestUtils
+
+from pyspark.sql import SparkSession, Row
+from pyspark.sql.types import StructType, StructField, StringType
+
+if have_pandas:
+    from pyspark.sql.connect.session import SparkSession as RemoteSparkSession
+from pyspark.sql.dataframe import DataFrame
+from pyspark.testing.connectutils import should_test_connect, connect_requirement_message
+from pyspark.testing.pandasutils import PandasOnSparkTestCase
```

Comment:
Hi, @zhengruifeng . This should be inside `if have_pandas`. I made a PR to fix it.
- https://github.com/apache/spark/pull/38929